plasmapy/classes/plasma.py

```diff
@@ -306,3 +310,48 @@ def alfven_speed(self):
         """
 
         return self.magnetic_field_strength / np.sqrt(mu0 * self.density)
+
+    @u.quantity_input(max_time=u.s)
+    def simulate(self, max_its=np.inf, max_time=np.inf*u.s):
+        """Simulates the plasma as set up, either for the given number of
+        iterations or until the simulation reaches the given time.
+
+        Parameters
+        ----------
+        max_its : int
+            Tells the simulation to run for a set number of iterations.
+
+        max_time : astropy.units.Quantity
+            Maximum total (in-simulation) time to allow the simulation to run.
+            Must have units of time.
+
+        Examples
+        --------
+        >>> # Run a simulation for exactly one thousand iterations.
+        >>> myplasma.simulate(max_time=1000)
+
+        >>> # Run a simulation for up to half an hour of simulation time.
+        >>> myplasma.simulate(max_time=30*u.minute)
+        """
+        if np.isinf(max_its) and np.isinf(max_time.value):
+            raise ValueError("Either max_time or max_its must be set.")
+
+        if np.isinf(max_time):
+            pb = ProgressBar(max_its)
```

Comment:
Could do, but the question then becomes how to procede in a predictable way if there are two possible end 'times'. Stop at whichever one is sooner I suppose, but I feel like that's going to lead to unexpected behaviour if users aren't careful. I'm happy to be convinced otherwise though.